File: 274. H-Index.cpp
class Solution {
public:
    int hIndex(vector<int>& citations) {
        vector<int>& a = citations;
        sort(a.begin(), a.end());
        int n = a.size();
        for (int i = 0; i < n; i++) {
            if (a[i] >= n-i)
                return n-i;
        }
        return 0;
    }
};

File: go/src/leetcode/274. H-Index/274. H-Index.go
package main

import (
	"fmt"
	"sort"
)

func hIndex(citations []int) int {
	n := len(citations)
	sort.Ints(citations)
	for i := range citations {
		if citations[i] >= n-i {
			return n-i
		}
	}
	return 0
}

func main() {
	fmt.Println(hIndex([]int{3,0,6,1,5}))
}
